server/models/Tier.js

```diff
@@ -71,6 +75,28 @@ export default function(Sequelize, DataTypes) {
   }, {
     paranoid: true,
 
+    instanceMethods: {
+      availableQuantity() {
+        return Sequelize.models.Response.sum('quantity', { 
+            where: {
+              TierId: this.id
+            }
+          })
+          .then(usedQuantity => {
+            if (this.maxQuantity && usedQuantity) {
+              return this.maxQuantity - usedQuantity;
+            } else if (this.maxQuantity) {
+              return this.maxQuantity;
+            } else {
+              return -1; // means there was no maxQuantity set, so infinite availability
```

Comment:
This might be confusing because for concurrency reasons, you may end up with a negative value from line 87.

so I would return `Infinity`;